src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -420,6 +440,7 @@ protected void retrieve(@CheckForNull SCMSourceCriteria criteria,
             if (context.wantTags()) {
                 referenceTypes.add(GitSCMTelescope.ReferenceType.TAG);
             }
+            //TODO DiscoverOtherRefsTrait?
```

Comment:
rainy day comment? (either delete it or if important but not blocking file a JIRA and update the comment with the JIRA)

Reply:
this is a comment for you I think, I sprinkled some of those comments around where they might be needed and then implemented in those places I definitely knew they where needed and I knew what to do.
I might have missed removing some that I implemented.